### ISSUE TITLE
RPC issue

### ISSUE THREAD

**Issue by cedricwalter** [kylemanna/docker-bitcoind#42]
If i run according to your config.md

`docker run -v bitcoind-data:/bitcoin --name=bitcoind-node -d \
        -p 8333:8333 \
        -p 127.0.0.1:8332:8332 \
        -e DISABLEWALLET=1 \
        -e PRINTTOCONSOLE=1 \
        -e RPCUSER=rpcuser \
        -e RPCPASSWORD=rpcpass \
        kylemanna/bitcoind`

or with
`docker run -v bitcoind-data:/bitcoin --name=bitcoind-node -d -p 8333:8333 -p 127.0.0.1:8332:8332 -v /etc/bitcoind/bitcoin.conf kylemanna/bitcoind`

/etc/bitcoind/bitcoin.conf  content being 
`rpcuser=rpcuser`
`rpcpassword=rpcpass`
`rpcport=8332`
`server=1`
`rpcbind=127.0.0.1`

i never succeed to connect with RPC
curl --user rpcuser --data-binary '{"jsonrpc": "1.0", "id":"curltest", "method": "getinfo", "params": [] }' -H 'content-type: text/plain;' http://127.0.0.1:8332/

always get a timeout

went inside the container 
`docker exec -it bitcoind-node bash`
and found
```
cat /bitcoin/.bitcoin/bitcoin.conf
disablewallet=1
printtoconsole=1
rpcuser=bitcoinrpc
rpcpassword=QvZjzT9o4ZkpgIMK4CyQDrRzGat1Bn57mKnJarne+0k8
```
Then i tried also 
`curl --user bitcoinrpc --data-binary '{"jsonrpc": "1.0", "id":"curltest", "method": "getinfo", "params": [] }' -H 'content-type: text/plain;' http://127.0.0.1:8332/`
I am a bit lost, did you try to use another rpc password already?

thanks and regards

**Comment by cedricwalter** [comment]
also tried to pass argument directly to process

docker run -v bitcoind-data:/bitcoin --name=bitcoind-node -d -p 8333:8333 -p 8332:8332 amacneil/bitcoin bitcoind -server -rpcallowip=127.0.0.1 -rpcuser=rpcuser -rpcpassword=rpcpass && docker logs -f bitcoind-node

saw a warning in logs 

2017-02-09 21:59:07 Config options rpcuser and rpcpassword will soon be deprecated. Locally-run instances may remove rpcuser to use cookie-based auth, or may be replaced with rpcauth. Please see share/rpcuser for rpcauth auth generation.

but should still work with latest Bitcoin version v0.13.0

**Comment by brandomr** [comment]
I noticed this too. The offending lines are 

https://github.com/kylemanna/docker-bitcoind/blob/167a4e63bebe216c8ad0527693b2bc8cfeacf2ac/bin/btc_init#L12-L17

You can see that a random password is set in the bitcoin.conf.

**Comment by mrbianchi** [comment]
same issue , testnet or mainnet, any port, additional data, the logs says:

> Binding RPC on address :: port 18332 failed.

**Comment by mrbianchi** [comment]
maybe this is the problem?
https://bitcointalk.org/index.php?topic=3179045.msg32928282#msg32928282

**Comment by mrbianchi** [comment]
https://github.com/kylemanna/docker-bitcoind/blob/master/docs/debug.md#running-bash-in-docker-container
> Note: This container will be run in the same way as the bitcoind node, but will not connect to already running containers or processes.

is related?

**Comment by mrbianchi** [comment]
/ruimarinho/bitcoin-core/ works well

**Comment by abhyankar-amey** [comment]
I am trying from a while with 0.18.1 bitcoin-core version but RPC not working. 

My commands =

1)
Start docker conainer = docker run --rm --name bitcoin-server -it btc-19nov
-printtoconsole
-regtest=1
-rpcbind=127.0.0.1
-rpcauth=

2)
Initiate RCP call from base Ubuntu 18.04 LTS machine = docker run -it --rm btc-19nov
bitcoin-cli
-rpcconnect=bitcoin-server
-regtest
-rpcuser=amey
-stdinrpcpass
getbalance

Error = error: Could not connect to the server bitcoin-server:18443

Make sure the bitcoind server is running and that you are connecting to the correct RPC port.

Any hints how to solve this issue?

I am using = Docker version 19.03.4, build 9013bf583a
Base OS = Ubuntu 18.04 LTS

If I telnet to localhost using the bind port, I get message connection refused.
For other docker app if I telnet on different port, it works!

**Comment by kylemanna** [comment]
This may have been my fault and should be fixed as of #72 and #73. 

Try this to launch a daemon with the name `mybitcoind`
```
$ docker run --rm -it -p 8332:8332 --name mybitcoind kylemanna/bitcoind
```

Simplest way to test is to have the `bitcoin-cli` on the host machine which can connect over 127.0.0.1:8332

Alternative you can use the client built in to the image, but need to look-up the remote address of the `bitcoind` daemon started above:
```
$ bitcoin-cli -rpcuser=bitcoinrpc -rpcpassword=$RPCPASSWORD  -rpcconnect=$(docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' mybitcoind) getrpcinfo
{
  "active_commands": [
    {
      "method": "getrpcinfo",
      "duration": 21
    }
  ],
  "logpath": "/bitcoin/.bitcoin/debug.log"
}

```

Let me know if there's still an issue.
